src/app/content/search/utils.ts

```diff
@@ -175,3 +183,18 @@ export const findSearchResultHit = (
 ): SearchResultHit | undefined => {
   return results.find((result) => result.source.elementId === target.elementId);
 };
+
+export const createRouteMatchOptions = (pageId: string, book: Book): AnyMatch | undefined => {
```

Comment:
this logic isn't actually specific to search, and overlaps with stuff in ContentLink i think, maybe some other places, you could move the util up and use it in more places